src/TSTransformer/nodes/statements/transformVariableStatement.ts

```diff
@@ -159,9 +128,28 @@ export function transformVariableDeclaration(
 		// in destructuring, rhs must be executed first
 		assert(node.initializer && value);
 		const name = node.name;
+
+		// optimize empty destructure
+		if (name.elements.length === 0) {
+			if (!luau.isArray(value) || !luau.list.isEmpty(value.members)) {
+				luau.list.pushList(statements, wrapExpressionStatement(value));
+			}
+			return statements;
+		}
+
 		if (ts.isArrayBindingPattern(name)) {
-			if (luau.isCall(value) && isLuaTupleType(state)(state.getType(node.initializer))) {
-				luau.list.pushList(statements, transformLuaTupleDestructure(state, name, value));
+			if (
+				luau.isCall(value) &&
+				isLuaTupleType(state)(state.getType(node.initializer)) &&
```

Comment:
oh by the way, these checks are wrong for some cases with unions and similar. They should use isDefinitelyType like the usual ones do. I think I covered it in #1829, or maybe in #1802 (the big one)